engine/crates/engine-v2/src/operation/bind/selection_set.rs

```diff
@@ -251,14 +289,62 @@ impl<'schema, 'p, 'binder> SelectionSetBinder<'schema, 'p, 'binder> {
         &mut self,
         parent: SelectionSetType,
         Positioned { node: fragment, .. }: &'p Positioned<engine_parser::types::InlineFragment>,
+        parent_directive_value_ids: &(&[QueryInputValueId], &[QueryInputValueId]),
     ) -> BindResult<()> {
         let ty = fragment
             .type_condition
             .as_ref()
             .map(|condition| self.bind_selection_set_type(parent, condition))
             .transpose()?
             .unwrap_or(parent);
-        self.register_selection_set_fields(ty, &fragment.selection_set)
+
+        let (mut skip_input_value_ids, mut include_input_value_ids) =
+            self.directives_to_value_ids(&fragment.directives)?;
+
+        skip_input_value_ids.extend_from_slice(parent_directive_value_ids.0);
+
+        include_input_value_ids.extend_from_slice(parent_directive_value_ids.1);
+
+        self.register_selection_set_fields(
+            ty,
+            &fragment.selection_set,
+            (&skip_input_value_ids, &include_input_value_ids),
+        )
+    }
+
+    fn directives_to_value_ids(
```

Comment:
would name `bind_executable_directives` or smth like this imho with the proposed struct a bit before

Reply:
Done